content/en/docs/unit-testing.md

```diff
@@ -0,0 +1,197 @@
+---
+title: "Unit Testing with the AWS SDK for Go V2"
+linkTitle: "Testing"
+description: "How to mock the AWS SDK for Go V2 when unit testing your application."
+---
+
+You can mock out the AWS SDK for Go V2 when unit testing your application by
+using Go interfaces. Using interface definitions you define the set of
+operations required by your application, and provide mock implementations of
+this interface when unit testing. You can follow this pattern to unit testing
+service client operations, paginators, and waiters.
+
+## Mocking Client Operations
+
+In this example, `S3GetObjectAPI` is an interface that defines the set of
+{{% alias service=S3 %}} API operations required by the `GetObjectFromS3`
+function. `s3GetObjectAPI` is satisfied by the {{% alias service=S3 %}}
+client's [GetObject]({{< apiref "service/s3#Client.GetObject" >}}) method.
+
+```go
+import "context"
+import "github.com/aws/aws-sdk-go-v2/service/s3"
+
+// ...
+
+type S3GetObjectAPI interface {
+	GetObject(ctx context.Context, params *s3.GetObjectInput, optFns ...func(*s3.Options)) (*s3.GetObjectOutput, error)
+}
+
+func GetObjectFromS3(api S3GetObjectAPI, bucket, key string) ([]byte, error) {
+	object, err := api.GetObject(context.TODO(), &s3.GetObjectInput{
+		Bucket: &bucket,
+		Key:    &key,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	all, err := ioutil.ReadAll(object.Body)
+	if err != nil {
+		return nil, err
+	}
+	return all, nil
+}
+```
+
+To test the `GetObjectFromS3` function, `mockGetObjectAPI` is used to satisfy
+the `s3GetObjectAPI` interface definition. The mock type is then to mock output
+and error responses returned from the service client.
+
+```go
+import "testing"
+import "github.com/aws/aws-sdk-go-v2/service/s3"
+
+// ...
+
+type mockGetObjectAPI func(ctx context.Context, params *s3.GetObjectInput, optFns ...func(*s3.Options)) (*s3.GetObjectOutput, error)
+
+func (m mockGetObjectAPI) PutObject(ctx context.Context, params *s3.GetObjectInput, optFns ...func(*s3.Options)) (*s3.GetObjectOutput, error) {
```

Comment:
This doesn't satisfy the interface anymore. This should be changed to `GetObject` too.